### PR TITLE
MCH custom electronics mapping fixes

### PR DESCRIPTION
**Pull request by aferrero2707**
Some bug fixes in the custom mapping initialization, plus a small code speed-up in the base eLink decoder.

### CODE REVIEW

File: Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx [hunk 1]
```diff
@@ -145,33 +145,34 @@ class DataDecoderTask
   }
 
  private:
-  std::string readFileContent(std::string_view filename)
+  std::string readFileContent(std::string& filename)
```
Comment (aphecetche) [inline Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx:148]: why the change from string_view to (non-const) string reference (which seems wrong) ?

**Comment by aphecetche** [comment]
concerning the small speed-up, do you have a measure of the gain ? (and a reproducer ?)

File: Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx (hunk 1 above)
Comment (aferrero2707) [reply Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx:148]: The code was not compiling when I tested it on our reference FLP. I will check again after a full O2Suite update...

**Comment by aferrero2707** [comment]
> concerning the small speed-up, do you have a measure of the gain ? (and a reproducer ?)

I do not have number to provide, but I checked with valgrind's cachegrind and noticed that the `sampaSync` constructor in
```
const uint64_t sync = sampaSync().uint64();
```
was one of the main cycle consumers, since the function gets called very often. I can provide a more quantitative answer if needed.

File: Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx (hunk 1 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx:148]: no, no need, string_view is indeed wrong in that context. Noticed it in #3461 also and then I went to const std::string& (and forced-push)

**Comment by aphecetche** [comment]
@aferrero2707 ok, the fix in custom mapping init should be in with #3461. For the speedup it will go into another, but in a slightly different form probably (there's a sampaSyncWord constant defined in SampaHeader.h that could be used instead of having a data member)
